pkg/controller/external_secrets_manager/controller.go

```diff
@@ -0,0 +1,247 @@
+	ControllerName = "external-secrets-manager"
+
+	// finalizer name for external-secrets.openshift.operator.io resource.
+	finalizer = "externalsecretsmanager.openshift.operator.io/" + ControllerName
+)
+
+var (
+	externalSecretsControllerId = fmt.Sprintf("externalsecrets.%s/%s", operatorv1alpha1.GroupVersion.Group, operatorv1alpha1.GroupVersion.Version)
+)
+
+// Reconciler reconciles externalsecretsmanager.openshift.operator.io CR.
+type Reconciler struct {
+	operatorclient.CtrlClient
+	Scheme          *runtime.Scheme
+	ctx             context.Context
+	eventRecorder   record.EventRecorder
+	log             logr.Logger
+	now             *common.Now
+	externalSecrets *operatorv1alpha1.ExternalSecrets
+}
+
+// +kubebuilder:rbac:groups=operator.openshift.io,resources=externalsecretsmanagers,verbs=get;list;watch;create;update;patch;delete
+// +kubebuilder:rbac:groups=operator.openshift.io,resources=externalsecretsmanagers/status,verbs=get;update;patch
+// +kubebuilder:rbac:groups=operator.openshift.io,resources=externalsecretsmanagers/finalizers,verbs=update
+
+// New is for building the reconciler instance consumed by the Reconcile method.
+func New(ctx context.Context, mgr ctrl.Manager) *Reconciler {
+	return &Reconciler{
+		CtrlClient:    NewClient(mgr),
+		Scheme:        mgr.GetScheme(),
+		ctx:           ctx,
+		eventRecorder: mgr.GetEventRecorderFor(ControllerName),
+		log:           ctrl.Log.WithName(ControllerName),
+		now:           &common.Now{},
+	}
+}
+
+func NewClient(m manager.Manager) operatorclient.CtrlClient {
+	return &operatorclient.CtrlClientImpl{
+		Client: m.GetClient(),
+	}
+}
+
+// SetupWithManager is for creating a controller instance with predicates and event filters.
+func (r *Reconciler) SetupWithManager(mgr ctrl.Manager) error {
+	statusUpdatePredicate := predicate.Funcs{
+		UpdateFunc: func(e event.UpdateEvent) bool {
+			oldObj := e.ObjectOld.(*operatorv1alpha1.ExternalSecrets)
+			newObj := e.ObjectNew.(*operatorv1alpha1.ExternalSecrets)
+			return !reflect.DeepEqual(oldObj.Status, newObj.Status)
+		},
+	}
+
+	return ctrl.NewControllerManagedBy(mgr).
+		For(&operatorv1alpha1.ExternalSecretsManager{}).
+		Named(ControllerName).
+		Watches(&operatorv1alpha1.ExternalSecrets{}, &handler.EnqueueRequestForObject{}, builder.WithPredicates(statusUpdatePredicate)).
+		Complete(r)
+}
+
+// Reconcile is the reconciliation loop to manage the current state of externalsecretsmanager CR.
+func (r *Reconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
+	r.log.V(1).Info("reconciling", "request", req)
+
+	// Fetch the externalsecretsmanager.openshift.operator.io CR
+	esm := &operatorv1alpha1.ExternalSecretsManager{}
+	key := types.NamespacedName{
+		Name: common.ExternalSecretsManagerObjectName,
+	}
+	if err := r.Get(ctx, key, esm); err != nil {
+		r.now.Do(func() {
+			r.eventRecorder.Eventf(esm, corev1.EventTypeWarning, "Read", "failed to fetch externalsecretsmanager.openshift.operator.io %q", key)
```

Comment:
Is this to avoid too many repeated events being logged at the same time (parallel reconcile doesn't seem enabled though?)